File: leetcode3/변지협/v3/3774. Absolute Difference Between Maximum and Minimum K Elements.py

'''
1. 아이디어 :
그냥 조건대로 풀면 된다.

2. 시간복잡도 :
o(n logn - 정렬)

3. 자료구조/알고리즘 :
'''

class Solution:
    def absDifference(self, nums: List[int], k: int) -> int:
        nums.sort()
        
        return sum(nums[::-1][:k]) - sum(nums[:k])

File: leetcode3/염혜정/3774. Absolute Difference Between Maximum and Minimum K Elements.java
// 오름차순 정렬 후 0 ~ k / nums.length - k
// O(n)

class Solution {
    public int absDifference(int[] nums, int k) {
        Arrays.sort(nums);
        int large = 0;
        int small = 0;

        for (int i = 0; i< k; i++) large += nums[i];
        for (int i = nums.length-1; i>=nums.length-k; i--) small += nums[i];

        return (int) Math.abs(large - small);
    }
}
